File: app/src/main/java/com/ipd/taxiu/ui/activity/balance/AddBankCardActivity.java
package com.ipd.taxiu.ui.activity.balance;

import android.os.Bundle;
import android.view.View;
import android.widget.TextView;

import com.ipd.taxiu.R;
import com.ipd.taxiu.ui.BaseUIActivity;
import com.ipd.taxiu.widget.MessageDialog;
import com.ipd.taxiu.widget.PickerUtil;

import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

import java.util.ArrayList;
import java.util.List;

/**
 * Created by Miss on 2018/7/30
 * 添加银行卡
 */
public class AddBankCardActivity extends BaseUIActivity implements View.OnClickListener {
    private TextView tv_choose_bank_type;
    private  List<String> data;
    private TextView btn_save;

    @Override
    protected int getContentLayout() {
        return R.layout.activity_add_bank_card;
    }

    @Override
    protected void initView(@Nullable Bundle bundle) {
        initToolbar();
        tv_choose_bank_type = findViewById(R.id.tv_choose_bank_type);
        btn_save = findViewById(R.id.btn_save);
    }

    @Override
    protected void loadData() {
        iniaData();
    }

    @Override
    protected void initListener() {
        tv_choose_bank_type.setOnClickListener(this);
        btn_save.setOnClickListener(this);
    }

    @NotNull
    @Override
    protected String getToolbarTitle() {
        return "添加银行卡";
    }

    @Override
    public void onClick(View v) {
        switch (v.getId()) {
            case R.id.tv_choose_bank_type:
                PickerUtil pickerUtil = new PickerUtil();
                pickerUtil.initBankCardOption(this, data, tv_choose_bank_type);
                break;
            case R.id.btn_save:
               toastShow("添加成功");
               finish();
                break;
        }
    }


    private   void iniaData() {
        data = new ArrayList<>();
        data.add("中国银行");
        data.add("中国建设银行");
        data.add("中国农业银行");
        data.add("中国工商银行");
        data.add("中国交通银行");
    }
}

File: app/src/main/java/com/ipd/taxiu/ui/activity/balance/AddBankCardActivity.kt
package com.ipd.taxiu.ui.activity.balance

import android.app.Activity
import android.content.Intent
import android.os.Bundle
import com.ipd.taxiu.R
import com.ipd.taxiu.ui.BaseUIActivity
import com.ipd.taxiu.utils.BankCardUtils
import com.ipd.taxiu.widget.BankCardTextWatcher
import com.ipd.taxiu.widget.PickerUtil
import kotlinx.android.synthetic.main.activity_add_bank_card.*

/**
Created by Miss on 2018/8/10
添加银行卡
 */
class AddBankCardActivity : BaseUIActivity() {
    private var bankType: Int = 0
    private val list = arrayListOf<String>("中国银行",
            "中国建设银行",
            "中国农业银行",
            "中国工商银行",
            "中国交通银行")
    private val pickerUtil = PickerUtil()

    companion object {
        fun launch(activity: Activity, bankType: Int) {
            val intent = Intent(activity, AddBankCardActivity::class.java)
            intent.putExtra("bankType", bankType)
            activity.startActivity(intent)
        }
    }

    override fun getContentLayout(): Int = R.layout.activity_add_bank_card
    override fun getToolbarTitle(): String{
        bankType = intent.getIntExtra("bankType", 0)
        if (bankType == MyBalanceActivity.ADD_BANK_CARD) {
            return "添加银行卡"
        }
        return  "修改银行卡"
    }

    override fun initView(bundle: Bundle?) {
        initToolbar()
    }

    override fun loadData() {
        BankCardTextWatcher.bind(et_bank_card_code)
    }

    override fun initListener() {
        tv_choose_bank_type.setOnClickListener { pickerUtil.initBankCardOption(this, list, tv_choose_bank_type) }
        btn_save.setOnClickListener {
            var bankCard:String = et_bank_card_code.text.toString().trim()
            if (!bankCard.isEmpty()) {
                bankCard = bankCard.replace(" ", "")
            }
            if (!BankCardUtils.checkBankCard(bankCard)){
                toastShow("请填写正确的银行卡号")
                return@setOnClickListener
            }
            if (bankType == MyBalanceActivity.ADD_BANK_CARD) {
                toastShow("添加成功")
            }
            if (bankType == MyBalanceActivity.UPDATE_BANK_CARD){
                toastShow("修改成功")
            }
            finish()
        }
    }
}